lms/djangoapps/oauth_dispatch/adapters/dop.py

```diff
@@ -0,0 +1,74 @@
+"""
+Adapter to isolate django-oauth-toolkit dependencies
+"""
+
+from provider.oauth2 import models
+from provider import constants, scope
+
+
+class DOPAdapter(object):
+    """
+    Standard interface for working with django-oauth-toolkit
+    """
+
+    backend = object()
+
+    def create_confidential_client(self, user, client_id=None):
+        """
+        Create an oauth client application that is confidential.
+        """
+        return models.Client.objects.create(
+            user=user,
+            client_id=client_id,
+            redirect_uri='http://example.edx/redirect',
```

Comment:
Because it's currently only used for tests that don't actually need this value.  I'll refactor.